crates/shim-kvm/src/snp/cpuid_page.rs

```diff
@@ -146,12 +144,19 @@ pub fn cpuid_count(leaf: u32, sub_leaf: u32) -> CpuidResult {
             .iter()
             .find_map(|e| {
                 if e.eax_in == leaf && e.ecx_in == sub_leaf {
-                    Some(CpuidResult {
+                    let mut res = CpuidResult {
                         eax: e.eax,
                         ebx: e.ebx,
                         ecx: e.ecx,
                         edx: e.edx,
-                    })
+                    };
+
+                    if leaf == 1 && sub_leaf == 0 && (res.ecx & (1u32 << 26)) != 0 {
```

Comment:
Is whatever this operation is doing properly documented by the existing function docstring? If not, can we add documentation explaining what this is doing?

Reply:
added docs and reshuffled things a bit.